internal/controllers/nmc_reconciler.go

```diff
@@ -1098,17 +1090,21 @@ func (p *podManagerImpl) baseWorkerPod(ctx context.Context, nmc client.Object, i
 			NodeName:           nodeName,
 			RestartPolicy:      v1.RestartPolicyOnFailure,
 			ServiceAccountName: item.ServiceAccountName,
-			Volumes:            append(volumes, psv...),
+			Volumes:            volumes,
 		},
 	}
 
-	if err = ctrl.SetControllerReference(nmc, &pod, p.scheme); err != nil {
+	if item.ImageRepoSecret != nil {
+		pod.Spec.ImagePullSecrets = []v1.LocalObjectReference{*item.ImageRepoSecret}
+	}
```

Comment:
How about we do this part before the pod template?

Something like:
```
var imagePullSecrets []v1.LocalObjectReference
if ... {
   imagePullSecrets = append(...)
}
```
 
and then add `ImagePullSecrets: imagePullSecrets` in the pod definition?

1. I think it is clearer because you see more stuff in the pod's definition rather than additional ifs after it
2. the field is set to `omitEmtpty` so an empty value won't show up

Reply:
Fixed